src/object/srv_obj_migrate.c

```diff
@@ -2517,14 +2522,19 @@ migrate_cont_iter_cb(daos_handle_t ih, d_iov_t *key_iov,
 	 * Open the remote container as a *client* to be used later to pull
 	 * objects
 	 */
+	if (daos_handle_is_valid(tls->mpt_cont_hdl))
+		dsc_cont_close(tls->mpt_pool_hdl, tls->mpt_cont_hdl);
```

Comment:
Hmm, this can not guarantee all object migration are finished?   Probably you can just do dsc_cont_open/close in migrate_obj_ult(), since cont_iter_cb() is on xstream 0, and migrate_obj_ult() might be in different ult. 

We can see if we can optimize this later.

Reply:
TLS destroyed when all ULTs complete. So, this should guarantee all object migration are finished.

[At the end](https://github.com/daos-stack/daos/blob/a665f2b27e423921d1f44fca63538d1f832013bd/src/object/srv_obj_migrate.c#L2324) of migrate_obj_ult() you can see migrate_pool_tls_put(tls) which will close pool and container.

Reply:
I mean when the dsc_cont_close(), the dc_cont will be freed.  but how do you know all those object migration ult (migrate_obj_ult) are done before dsc_cont_close.

IMHO, you probably need increase refcount for cont_hdl(dc_cont) before create migrate_obj_ult(), then release the refcount when migrate_obj_ult() is done.

Reply:
Probably you right. Maybe we should keep it open until completing all ULTs. The same looks for object handle. Need to elaborate this one more time.

Reply:
Are we going to fix it in current patch or a follow-on patch?

Reply:
@NiuYawei, I have to resolve somehow rebuild test fails. Otherwise this patch will not be landed.